### ISSUE TITLE
How to add layers to a pretrained model?

### ISSUE THREAD

**Issue by TamHHM** [ThilinaRajapakse/simpletransformers#750]
**Describe the bug**
I loaded the roberta-large-mnli model which has 3 output labels. I tried to add another linear layer of (3,2) at the bottom of Roberta classifier to reduce the prediction to just 2 labels . I then fit the model with 1 epoch to see the change. However, the prediction function still gave me a score vector of three.

**To Reproduce**
```
import pandas as pd
import torch
from simpletransformers.classification import ClassificationModel

train_args = {
    'output_dir': 'test/',
    'overwrite_output_dir': True,
    'num_train_epochs': 1,
    'fp16': False,
}
model = ClassificationModel('roberta', 'roberta-large-mnli',
                            num_labels=3, args=train_args)

for param in model.model.base_model.parameters():
    param.requires_grad = False
    
model.num_labels = 2
model.args.labels_list = [0,1]
model.model.classifier.add_module("contr_layer", torch.nn.modules.linear.Linear(3,2))

train_df = pd.DataFrame(dict(
    text_a = ["I love you"],
    text_b = ["I hate you"],
    labels = [0]
))

eval_df = pd.DataFrame(dict(
    text_a = ["I love cat"],
    text_b = ["I hate cat"],
    labels = [0]
))
model.train_model(train_df, eval_df=eval_df)
model.predict([["a","b"]])
```

**Expected behavior**
Expecting a score vector of two elements instead of three.

**Comment by ThilinaRajapakse** [comment]
Manually adding the extra layer to `model.classifier` doesn't mean that it's going to be used. The actual path through the `classifier` module is defined in its `forward()` function, as shown below.

```python
class RobertaClassificationHead(nn.Module):
    """Head for sentence-level classification tasks."""

    def __init__(self, config):
        super().__init__()
        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
        self.dropout = nn.Dropout(config.hidden_dropout_prob)
        self.out_proj = nn.Linear(config.hidden_size, config.num_labels)

    def forward(self, features, **kwargs):
        x = features[:, 0, :]  # take <s> token (equiv. to [CLS])
        x = self.dropout(x)
        x = self.dense(x)
        x = torch.tanh(x)
        x = self.dropout(x)
        x = self.out_proj(x)
        return x
```
To get it to work, off the top of my head, you would have to write your own `RobertaClassificationHead` and replace model.classifier with it. I'm not sure whether loading weights will work correctly after that though.

Regardless, the extra layer you add will be randomly initialized anyway.  I'm wondering why you'd go this route instead of just making the model a 2 label model though.

E.g.:

```python
import pandas as pd
import torch
from simpletransformers.classification import ClassificationModel

train_args = {
    "output_dir": "test/",
    "overwrite_output_dir": True,
    "num_train_epochs": 1,
    "fp16": False,
}
model = ClassificationModel("roberta", "roberta-base", num_labels=2, args=train_args)

train_df = pd.DataFrame(dict(text_a=["I love you"], text_b=["I hate you"], labels=[0]))

eval_df = pd.DataFrame(dict(text_a=["I love cat"], text_b=["I hate cat"], labels=[0]))
model.train_model(train_df, eval_df=eval_df)
model.predict([["a", "b"]])

```

**Comment by TamHHM** [comment]
Thanks for the reply. I figured out that I have to manually create my own classifier as you suggested. Then copy the weights from to previous on (before being replaced) to the new one. The reason I did this is because I wanted to start with a pretrained model "roberta-large-mnli" which has 3 layers defined in the classifier. If I put two as you suggested, an error will pop up.

**Comment by ThilinaRajapakse** [comment]
```python
import pandas as pd
import torch
from simpletransformers.classification import ClassificationModel

train_args = {
    "output_dir": "test/",
    "overwrite_output_dir": True,
    "num_train_epochs": 1,
    "fp16": False,
}
model = ClassificationModel("roberta", "roberta-base", num_labels=2, args=train_args)

train_df = pd.DataFrame(dict(text_a=["I love you"], text_b=["I hate you"], labels=[0]))

eval_df = pd.DataFrame(dict(text_a=["I love cat"], text_b=["I hate cat"], labels=[0]))
model.train_model(train_df, eval_df=eval_df)
model.predict([["a", "b"]])
```

This runs without any errors. You can use any ClassificationModel with any number of output labels. The classification layer will be randomly initialized, but it shouldn't throw any errors.

**Comment by TamHHM** [comment]
Thanks again for your response. You missed that I loaded "roberta-large-mnli" instead of "roberta-base". I had no problem loading "roberta-base".

**Comment by ThilinaRajapakse** [comment]
Ah yes, I missed that, my bad.

**Comment by stefanolipira** [comment]
How is the classification model with labels=2 compiled? with a categorical cross-entropy loss function and categorical accuracy as the evaluation metric?
